### DISCUSSION TITLE
Where is the "Images" in render template?

### DISCUSSION THREAD

**Discussion by guguxh** [wxyz-abcd/node-haxball#47] [General]
I don't see this in any place

**Comment by wxyz-abcd** [comment]
I will write an example file for renderer very soon. I will upload the images with that example. For now, you can experiment with random images. The object structure has 4 keys: `grass`, `concrete`, `concrete2`, `typing`. Each of them should have an `Image` object as its value. You may create these `Image` objects by either using `createImageBitmap` or as html `<img />` elements.

**Reply by guguxh** [reply]
Alright I'll try this. But the object isn't already created? In the modules.export you export the "images".

**Reply by wxyz-abcd** [reply]
No. You have to import the renderer with a custom name(such as `DefaultRenderer`) first. Then, in order to use it, initialize it by using its constructor: `new DefaultRenderer(API, localStorageObj, canvas, images, renderApi)`, meaning that these 5 parameters should be initialized by yourself first. The last step is using this renderer object with `Room.join` or `room.setRenderer`.

**Reply by guguxh** [reply]
Oh, I get it. Thanks.

**Reply by wxyz-abcd** [reply]
I just updated the renderer interface. It's much easier now. :)

**Comment by wxyz-abcd** [comment]
Every web resource is available now at `examples_web` folder.

**Reply by guguxh** [reply]
I see. But I think this isn't working right. When I start the server, the site don't show the roomlist exactly.

**Reply by wxyz-abcd** [reply]
It does work correctly. If it does not, you probably don't use the browser extension or there's some other problem.

**Reply by wxyz-abcd** [reply]
What problem do you have exactly?

**Reply by guguxh** [reply]
> What problem do you have exactly?
![image](https://user-images.githubusercontent.com/61206153/213479198-8a14fe99-5504-4835-842a-895a528320f0.png)
This happen.

**Reply by wxyz-abcd** [reply]
It happens because you don't use haxballOriginModifier extension. It's already written there that you need that extension to work with this example. There are instructions about how to use it in its main folder in this repo.

**Reply by wxyz-abcd** [reply]
And please, read the README.md file in the examples_web folder first.

**Reply by guguxh** [reply]
> It happens because you don't use haxballOriginModifier extension. It's already written there that you need that extension to work with this example. There are instructions about how to use it in its main folder in this repo.

I use, but I think it's because isn't updated.

**Reply by wxyz-abcd** [reply]
You can try to look at the console of your browser's developer tools just after you click on "refresh" button. If the error is a cross-domain network error, which is the case, i'm almost certain of that, your extension configuration has to be wrong. Maybe a screenshot of the error might help me solve your problem.

**Reply by wxyz-abcd** [reply]
I added you to discord with the hope that we will solve the issues together :)
